DependencyInjection/HttplugExtension.php

```diff
@@ -90,4 +92,66 @@ protected function configureClients(ContainerBuilder $container, array $config)
             $container->setAlias('httplug.client.default', 'httplug.client.'.$first);
         }
     }
+
+    /**
+     * @param ContainerBuilder $container
+     * @param array            $config
+     */
+    protected function configurePlugins(ContainerBuilder $container, array $config)
+    {
+        foreach ($config as $name => $pluginConfig) {
+            $pluginId = 'httplug.plugin.'.$name;
+            if ($pluginConfig['enabled']) {
+                $def = $container->getDefinition($pluginId);
+                $this->configurePluginByName($name, $def, $pluginConfig);
```

Comment:
I was considering that approach. Switch statements are code smell but dynamic functions like that is worse. Also, you cant type hint dynamic functions in PHP7